src/Mod/Draft/draftobjects/draftlink.py

```diff
@@ -0,0 +1,180 @@
+# ***************************************************************************
+# *   Copyright (c) 2009, 2010 Yorik van Havre <yorik@uncreated.net>        *
+# *   Copyright (c) 2009, 2010 Ken Cline <cline@frii.com>                   *
+# *   Copyright (c) 2020 FreeCAD Developers                                 *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+"""This module provides the object code for the Draft Link object.
+"""
+## @package draftlink
+# \ingroup DRAFT
+# \brief This module provides the object code for the Draft Link object.
+
+from PySide.QtCore import QT_TRANSLATE_NOOP
+
+import FreeCAD as App
+
+from draftutils.utils import get_param
+
+from draftobjects.base import DraftObject
+
+
+class DraftLink(DraftObject):
+    """
+    Documentation needed.
+    I guess the DraftLink was introduced by Realthunder to allow
+    the use of new App::Link object into Draft Array objects during
+    the development of version 0.19.
+    """
+
+    def __init__(self,obj,tp):
+        self.use_link = False if obj else True
+        super(DraftLink, self).__init__(obj, tp)
+        if obj:
+            self.attach(obj)
```

Comment:
I am not sure why the array code runs the attach method in the initialization. In the `array.Array` class it's the same thing. Instead of adding the properties in `__init__` they are added through `attach`.

Reply:
this is a pattern @realthunder used also in the arch object he provided me. I think this is done to init both object and viewprovider and after setting their properties... but i'm nowhere near certain about that...